src/plexil-adapter/OwInterface.cpp

```diff
@@ -450,55 +421,44 @@ double OwInterface::groundPosition () const
   return GroundPosition;
 }
 
-static void guarded_move_callback
-(const ow_lander::GuardedMoveResult::ConstPtr& msg)
+static void guarded_move_done_cb
+(const actionlib::SimpleClientGoalState& state,
+ const ow_lander::GuardedMoveResultConstPtr& result)
 {
-  if (msg->success) {
-    GroundFound = true;
-    string frame = msg->frame;
-    string valid = "base_link";
-    if (frame != valid) {  // the only supported value
-      ROS_ERROR("GuardedMoveResult frame was not %s", valid.c_str());
-      return;
-    }
-    GroundPosition = msg->position.z;
-    publish ("GroundFound", true);
-    publish ("GroundPosition", GroundPosition);
-  }
-  else {
-    ROS_WARN("GuardedMove did not find ground!");
-  }
+  ROS_INFO ("GuardedMove finished in state %s", state.toString().c_str());
+  GroundFound = result->success;
+  GroundPosition = result->final.z;
+  publish ("GroundFound", GroundFound);
+  publish ("GroundPosition", GroundPosition);
 }
 
+//////////////////// General Action support ///////////////////////////////
 
-//////////////////// GuardedMove Action support ////////////////////////////////
+const auto ActionTimeoutSecs   = 180.0; // TODO: make this action-specific
+const auto ActionServerTimeout = 10.0;  // seconds
 
-// At present, this is a prototypical ROS action using a dummy server in this
-// directory, GuardedMoveServer.  It is NOT connected to the testbed.
+//////////////////// ROS Action callbacks - generic //////////////////////
 
-static void guarded_move_done_cb
-(const actionlib::SimpleClientGoalState& state,
- const ow_autonomy::GuardedMoveResultConstPtr& result)
+template<typename T>
+static void action_feedback_cb (const T& feedback)
 {
-  ROS_INFO ("GuardedMove done callback: finished in state [%s]",
-            state.toString().c_str());
-  ROS_INFO("GuardedMove done callback: result (%f, %f, %f)",
-           result->final.x, result->final.y, result->final.z);
 }
 
-static void guarded_move_active_cb ()
+template<int OpIndex>
+static void active_cb ()
 {
-  ROS_INFO ("GuardedMove active callback - goal active!");
+  ROS_INFO ("%s started...", LanderOpNames[OpIndex].c_str());
 }
 
-static void guarded_move_feedback_cb
-(const ow_autonomy::GuardedMoveFeedbackConstPtr& feedback)
+template<int OpIndex, typename T>
+static void action_done_cb
+(const actionlib::SimpleClientGoalState& state,
+ const T& result_ignored)
```

Comment:
It doesn't seem like there is a need to templatize the second argument
```suggestion
template<int OpIndex>
static void action_done_cb
(const actionlib::SimpleClientGoalState& state,
 const ResultPtr& result_ignored)
```

Reply:
Your suggestion doesn't compile, as ResultPtr does not name a class.  I tried auto as well, not thinking that would work.

Reply:
I didn't pay attention that these are specified as arguments to the template method and thought they were the same classes that were used. Please disregard this comment and related ones